sdformat_to_mjcf/sdformat_to_mjcf/converters/link.py

```diff
@@ -43,10 +45,14 @@ def add_link(body, link, parent_name="world"):
     #     - light (TODO (azeey))
     #     - particle_emitter
     sem_pose = link.semantic_pose()
-    if parent_name == 'world':
-        pose = su.graph_resolver.resolve_pose(sem_pose)
+    if pose is None:
+        if parent_name == 'world':
+            pose = su.graph_resolver.resolve_pose(sem_pose)
+        else:
+            pose = su.graph_resolver.resolve_pose(sem_pose, parent_name)
     else:
-        pose = su.graph_resolver.resolve_pose(sem_pose, parent_name)
+        link_pose = su.graph_resolver.resolve_pose(link.semantic_pose())
+        pose = pose * link_pose
     body = body.add("body",
                     name=link.name(),
```

Comment:
No that we're supporting `<world>` which might have multiple models, I think we need to prefix the link name here to avoid name collisions.

Reply:
link names are not unique?

Reply:
> link names are not unique?

Not across models. Consider:
```xml
<world name="test_world">
  <model name="M1">
    <link name="base_link"/>
  </model>
  <model name="M2">
    <link name="base_link"/>
  </model>
</world>
```

Both models `M1` and `M2` have a link called `base_link`. This is valid in SDFormat because the namespace is defined by the xml hierarchy. In Mujoco, the namespace is defined by element type, so all bodies are in the same namespace, which would imply a name collision in this case.

Reply:
https://github.com/gazebosim/gz-mujoco/pull/37/commits/bdff1664ade8fca2b12eeb2d883f8c503dade1d7